pkg/resolution/resource/name.go

```diff
@@ -111,3 +121,28 @@ func GenerateDeterministicNameFromSpec(prefix, base string, resolutionSpec *v1be
 	}
 	return name[:strings.LastIndex(name[:maxLength], " ")], nil
 }
+
+// GenerateErrorLogString makes a best effort attempt to get the name of the task
+// when a resolver error occurred.  The TaskRef name does not have to be set, where
+// the specific resolver gets the name from the parameters.
+func GenerateErrorLogString(resolverType string, params v1.Params) string {
+	paramString := fmt.Sprintf("resolver type %s\n", resolverType)
+	for _, p := range params {
+		if p.Name == ParamName {
+			name := p.Value.StringVal
+			if p.Value.Type != v1.ParamTypeString {
+				asJSON, err := p.Value.MarshalJSON()
+				if err != nil {
+					paramString += fmt.Sprintf("name could not be marshalled: %s\n", err.Error())
+					continue
+				}
+				name = string(asJSON)
```

Comment:
NIT: coverage shows no coverage for this line, can we add a test case with an object param that can be marshalled as JSON? It can be a separate PR.

Reply:
Yep I'll handle this in separate PR ... hopefully should have it open today

Reply:
see https://github.com/tektoncd/pipeline/pull/7950